src/mscorlib/shared/System/Guid.Unix.cs

```diff
@@ -10,11 +10,12 @@ namespace System
     partial struct Guid
     {
         // This will create a new random guid based on the https://www.ietf.org/rfc/rfc4122.txt 
-        public static unsafe Guid NewGuid()
+        public static Guid NewGuid()
         {
-            Guid g;
-            Interop.GetRandomBytes((byte*)&g, sizeof(Guid));
-            
+            Span<byte> randomBytes = stackalloc byte[Marshal.SizeOf<Guid>()];
+            Interop.GetRandomBytes(randomBytes);
+            Guid g = new Guid(randomBytes);
```

Comment:
This extra copy of Guid maybe also slowing down things measurably. We care about performance of `NewGuid`.

Reply:
It would be nice to submit this refactoring as independent PR where we can focus on its correctness and performance effects.